src/parser.js

```diff
@@ -734,8 +736,29 @@ export default class Parser {
         transition.id = methodValue.operationId;
       }
 
+      /* eslint-disable prefer-const */
+      let resourceParametersWithPath = [];
+      /* eslint-enable prefer-const */
+
+      _.forEach(resourceParameters, (resourceParameter, index) => {
```

Comment:
Because `_.forEach` takes care of objects too unlike `for of` which can be finicky. So, we have standardised using `_.forEach`.